layer/src/test/scala/geotrellis/layer/LayoutTileSourceSpec.scala

```diff
@@ -268,13 +271,54 @@ class LayoutTileSourceSpec extends AnyFunSpec with RasterMatchers {
           .get
           .band(0)
         val arr = tile.toArray
-        val ones = tile.mapIfSet(i => 1).toArray()
-        ones.sum shouldBe(arr.size)
+        val ones = tile.mapIfSet(_ => 1).toArray()
+        ones.sum shouldBe arr.size
       }
 
       neighborhood map {
         case (x, y) => checkExtent(x, y, rs)
       }
     }
+
+    // https://github.com/locationtech/geotrellis/issues/3253
+    it("MosaicRasterSource should be properly tiled to layout") {
```

Comment:
The idea of the test, is to prove that the read by extent behaves the same way as the read by keys (by bounds).